### PR TITLE
Refactor/deduplicate using symlinks

### PR DESCRIPTION
**Pull request by ArturGajowy**
This is a prerequisite for the work on images for BigInsights / IBM Hadoop Open Platform distro

### CODE REVIEW

**Comment by ArturGajowy** [comment]
Presto-checks builds here:
https://travis-ci.org/Teradata/presto-checks/builds/168596600 [HDP]
https://travis-ci.org/Teradata/presto-checks/builds/168596559 [CDH]

Both builds passed :)

File: teradatalabs/centos6-ssh-oj8/Dockerfile [hunk 1]
```diff
@@ -20,9 +20,13 @@ ARG JDK_RPM
 ARG JDK_PATH
 
 # Install Oracle Java and presto-admin dependences (already has python 2.6)
-RUN yum upgrade ca-certificates --disablerepo=epel -y && \
-    yum update -y && \
-    yum install -y wget bzip2 gcc python-devel tar expect && \
+RUN yum install -y \
```
Comment (ebd2) [inline teradatalabs/centos6-ssh-oj8/Dockerfile:23]: Any idea why the cert and epel stuff changed? If the build and tests run, I believe that it works, but I'm curious.

File: teradatalabs/hdp2.5-base/Dockerfile [hunk 2]
```diff
@@ -15,23 +15,32 @@
...
-RUN yum install -y hadoop-hdfs-namenode \
-	&& yum install -y hadoop-hdfs-secondarynamenode \
-	&& yum install -y hadoop-hdfs-datanode
+  && yum install -y \
+    hadoop-hdfs-namenode \
+    hadoop-hdfs-secondarynamenode \
+    hadoop-hdfs-datanode \
 
-RUN yum install -y hadoop-mapreduce-historyserver
+    hadoop-mapreduce-historyserver \
```
Comment (ebd2) [inline teradatalabs/hdp2.5-base/Dockerfile:26]: Do all of these share a version number so we don't have any hope of getting a cache hit if only e.g. hive changed?

**Review by ebd2 — APPROVED** [review]
This is awesome. Let me know if you want me to take another look before you push.

**Comment by ebd2** [comment]
Please make sure you push a presto-admin branch with a snapshot build to make sure we don't break the images it relies on.

File: teradatalabs/centos6-ssh-oj8/Dockerfile (hunk 1 above)
Comment (ArturGajowy) [reply teradatalabs/centos6-ssh-oj8/Dockerfile:23]: No idea, I just saw that it works in other images and ~~let a steam-roller over~~ unified it :)

File: teradatalabs/hdp2.5-base/Dockerfile (hunk 2 above)
Comment (ArturGajowy) [reply teradatalabs/hdp2.5-base/Dockerfile:26]: I assume the guys at Hortonworks will update versions rarely and will keep them consistent :) In other words, my assumption is those do share a version number of 2.5.0.0

**Comment by cawallin** [comment]
I believe that those changes are needed for presto-admin.

**Comment by ArturGajowy** [comment]
Glad you like it! Presto-admin tests triggered here:
https://github.com/prestodb/presto-admin/pull/281
build results: https://travis-ci.org/prestodb/presto-admin/builds/169059308
presto vs cdh-latest-snapshot build: https://travis-ci.org/Teradata/presto-checks/builds/169028481
presto vs hdp-latest-snapshot build: https://travis-ci.org/Teradata/presto-checks/builds/169028590

All the builds haven't finished yet at the moment of writing this :)

**Comment by ArturGajowy** [comment]
@cawallin do you mean the epel repo removal, the certificates, or the package list? Because the package list is (effectively) the same (python-setuptools are inherited, pip was unused) and the former changes would blow up during docker build, I think?

EDIT: disregard the setuptools and pip part, that's a different change than we're discussing

**Comment by cawallin** [comment]
It could be unused now, I just remember adding those parts to make the
presto-admin tests happy, I haven't taken a look at the changeset. As long
as the presto-admin tests pass I'm satisfied :)

**Comment by ArturGajowy** [comment]
@cawallin cool :)

Presto-admin and presto vs cdh passed, presto vs hdp fails because of a stupid error in the last commit that I pushed a fixup for just now. Will upload the new snapshot and run vs hdp again.

**Comment by ArturGajowy** [comment]
Removed a stray iop-base image from 'Remove redundant package install from cdh5-master'

File: teradatalabs/cdh5-hive/Dockerfile [hunk 3]
```diff
@@ -16,32 +16,36 @@ FROM teradatalabs/cdh5-base
 MAINTAINER Teradata Docker Team <docker@teradata.com>
 
 # INSTALL HADOOP AND HIVE
-RUN yum install -y hadoop-conf-pseudo hive hive-server2 zookeeper-server
+RUN \
+  yum install -y \
```
Comment (kokosing) [inline teradatalabs/cdh5-hive/Dockerfile:20]: you could sort packages here

File: teradatalabs/centos6-ssh-oj8/Dockerfile [hunk 4]
```diff
@@ -21,7 +21,13 @@ ARG JDK_PATH
...
 # Install Oracle Java and presto-admin dependences (already has python 2.6)
 RUN yum upgrade ca-certificates --disablerepo=epel -y && \
-    yum install -y wget bzip2 gcc python-devel tar expect && \
+    yum install -y \
+      wget \
+      bzip2 \
+      gcc \
+      python-devel \
+      tar \
+      expect && \
```
Comment (kokosing) [inline teradatalabs/centos6-ssh-oj8/Dockerfile:29]: sort packages

File: commons/conf-setup/core-site.xml [hunk 5]
```diff
@@ -0,0 +1,74 @@
+<?xml version="1.0"?>
```
Comment (kokosing) [inline commons/conf-setup/core-site.xml:1]: I would squash all `Unifiy *` commits into single one

File: teradatalabs/cdh5-hive-kerberized/files/conf/yarn-site.xml [hunk 6]
```diff
@@ -15,7 +15,6 @@
   See the License for the specific language governing permissions and
   limitations under the License.
 -->
-<?xml-stylesheet type="text/xsl" href="configuration.xsl"?>
```
Comment (kokosing) [inline teradatalabs/cdh5-hive-kerberized/files/conf/yarn-site.xml:18]: `<?xml>` and `<?xml-stylesheet>` are different. Maybe you could merge this nodes into single `<?xml stylesheet..>`

**Review by kokosing — APPROVED** [review]
Well done!

File: teradatalabs/cdh5-hive-kerberized/files/conf/yarn-site.xml (hunk 6 above)
Comment (ArturGajowy) [reply teradatalabs/cdh5-hive-kerberized/files/conf/yarn-site.xml:18]: It seems most our config files don't contain this line while the books on hadoop have it in config excertps. So I assume this is redundant.

Reworded the commits to `Remove redundant <xml-stylesheet/> declaration (...)`

**Comment by ArturGajowy** [comment]
Reworded the `Remove duplicate <xml/> declaration (...)` commits to `Remove redundant <xml-stylesheet/> declaration (...)`

**Pull request merged**